Migrate Prometheus Module to Package

ChrsMark opened the pull request:
Implementing Prometheus package migration

Screenshots:
<img width="1530" alt="Screenshot 2020-06-29 at 20 20 32" src="https://user-images.githubusercontent.com/11754898/86036374-6e589e80-ba46-11ea-88f5-8ccd929b0bd0.png">
<img width="1680" alt="Screenshot 2020-06-29 at 20 22 54" src="https://user-images.githubusercontent.com/11754898/86036381-70226200-ba46-11ea-8e89-ec32d80c229c.png">
<img width="1530" alt="Screenshot 2020-06-29 at 20 22 28" src="https://user-images.githubusercontent.com/11754898/86036383-71538f00-ba46-11ea-86e5-5d93f4e64df0.png">
![Screenshot 2020-07-06 at 12 37 16](https://user-images.githubusercontent.com/11754898/86579061-786a2800-bf85-11ea-902a-b6ad3206fdf4.png)

elasticmachine commented:
## :green_heart: Build Succeeded


<!-- BUILD BADGES-->
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fintegrations%2FPR-114/detail/PR-114/1//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fintegrations%2FPR-114/detail/PR-114/1//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fintegrations%2FPR-114/detail/PR-114/1//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fintegrations%2FPR-114/detail/PR-114/1//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://integrations_114.docs-preview.app.elstc.co/diff)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats

* Build Cause: `[Pull request #114 opened]`

* Start Time: 2020-06-25T09:15:34.431+0000

* Duration: 3 min 58 sec





</p>
</details>

<!-- TEST RESULTS IF ANY-->


<!-- STEPS ERRORS IF ANY-->





<!--COMMENT_GENERATED_WITH_ID_comment.id-->

elasticmachine commented:
Pinging @elastic/integrations (Team:Integrations)

mtojek commented:
Please also rebase against master.

ChrsMark commented:
Thanks for the review @mtojek ! 

PR has been updated:

- Added screenshot
- Addressed comments and rebased
- Renamed `datasources` to `config_templates` 
- Renamed `metrics` category to `datastores`

exekias commented on packages/prometheus/dataset/collector/fields/fields.yml:
```diff
@@ -0,0 +1,11 @@
+- name: prometheus
```
This file seems to be missing some fields. I would expect it to be a mix between x-pack and oss fields.yml's, as we are shipping the basic agents with the agent

ChrsMark replied:
🤦 good catch, thanks!

ChrsMark commented:
@exekias I added manually the missing fields and tested manually again to verify nothing is broken. Feel free to have another look :)!
